README.md

```diff
@@ -227,18 +343,121 @@ NxtPipeline::Pipeline.new do |p|
 end
 ```
 
-Note that the `after_execute` callback will not be called in case a step raises an error. 
+Note that the `after_execute` callback will not be called in case a step raises an error.
 See the previous section (_Error callbacks_) for how to define callbacks that run in case of errors.
 
-### Step resolvers
+### Constructor resolvers
+
+You can also define constructor resolvers for a pipeline to dynamically define which previously registered constructor
+to use for a step based on the argument and options passed to the step.
+
+```ruby
+class Transform
+  def initialize(word, operation)
+    @word = word
+    @operation = operation
+  end
+
+  attr_reader :word, :operation
+
+  def call
+    word.send(operation)
+  end
+end
+
+NxtPipeline.new do |pipeline|
+  # dynamically resolve to use a proc as constructor
+  pipeline.constructor_resolver do |argument, **opts|
+    argument.is_a?(Class) &&
+      ->(step, arg:) {
+        result = step.argument.new(arg, opts.fetch(:operation)).call
+        # OR result = step.argument.new(arg, step.operation).call
+        { arg: result }
+      }
```

Comment:
With constructor resolvers you can say which constructor to use based on the step and argument. We could for instance say to always use :service when something inherits from NxtInit and always use :job in case something inherits from ApplicationJob. Then you would not have to provide the `constructor: :service` options anymore.

Reply:
But you can also directly return a constructor (something that responds to call)